core/contracts/financial_templates/Liquidatable.sol

```diff
@@ -320,12 +316,7 @@ contract Liquidatable is PricelessPositionManager {
             } else if (msg.sender == sponsor) {
                 // Pay SPONSOR: remaining collateral (locked collateral - TRV) + sponsor reward
                 FixedPoint.Unsigned memory remainingCollateral;
-                // if TRV > Collateral, there is 0 excess collateral
-                if (tokenRedemptionValue.isGreaterThan(liquidation.lockedCollateral)) {
-                    remainingCollateral = FixedPoint.fromUnscaledUint(0);
-                } else {
-                    remainingCollateral = liquidation.lockedCollateral.sub(tokenRedemptionValue);
-                }
+                remainingCollateral = liquidation.lockedCollateral.sub(tokenRedemptionValue);
```

Comment:
Two general comments about the way this is structured: what prevents some party (like the sponsor) from calling multiple times and getting their reward twice?

Reply:
will think about this and get back to you.

Reply:
There is one point where this could be a problem. I have added a todo.

Reply:
Sweet, thanks!